src/sheet_metal_conus.py

```diff
@@ -230,47 +232,40 @@ def effect(self):
 
         linestyle = { 'stroke' : self.options.strokeColour, 'fill' : 'none',
                       'stroke-width': str(self.getUnittouu(str(self.options.strokeWidth) + self.options.units)) }
-
-        # Connect the points with lines
-        self.drawLine(dictCone['ptA'], dictCone['ptB'], linestyle, 'lineAB', convFactor, grp)
-        self.drawLine(dictCone['ptC'], dictCone['ptD'], linestyle, 'lineCD', convFactor, grp)
-        # Connect the points with arcs:
-        # draw arc with center 0,0 
-        # - a circle is an ellipse with radiusX=radiusY !
+        line_attribs = {'style' : simplestyle.formatStyle(linestyle), inkex.addNS('label','inkscape') : 'Cone' }
+        
+        # Connect the points into a single path of lines and arcs
         zeroCenter=(0.0, 0.0)
-        self.drawEllipse(dictCone['shortRadius'], dictCone['shortRadius'], zeroCenter, 0.0, dictCone['angle'], linestyle, 'arcAD', convFactor, grp)
-        self.drawEllipse(dictCone['longRadius'],  dictCone['longRadius'],  zeroCenter, 0.0, dictCone['angle'], linestyle, 'arcBC', convFactor, grp  )
+        angle = math.degrees(dictCone['angle'])
+        path = ""
+        path += self.build_line(dictCone['ptA'], dictCone['ptB'], convFactor) # A,B
+        path += " " + self.build_arc(zeroCenter, 0.0, angle, dictCone['longRadius']*convFactor)
+        path += " " + self.build_line(dictCone['ptC'], dictCone['ptD'], convFactor) # C,D
+        path += self.build_arc(zeroCenter, angle, 0.0, dictCone['shortRadius']*convFactor, False)
+        line_attribs['d'] = path
+        ell = inkex.etree.SubElement(grp, inkex.addNS('path','svg'), line_attribs )
```

Comment:
Here we replace original four separate entities (line,lin, ellipse, ellipse) with one path.
Easier for user to select as a single entity instead of four separate paths.